spec/FrameSpec.js

```diff
@@ -0,0 +1,299 @@
+      expect(lastFrame.pinsRemaining()).toEqual(10);
+      lastFrame.play(10);
+      expect(lastFrame.pinsRemaining()).toEqual(10);
+      lastFrame.play(10);
+      expect(lastFrame.pinsRemaining()).toEqual(10);
+      lastFrame.play(10);
+      expect(lastFrame.pinsRemaining()).toEqual(0);
+    });
+
+    it("should be calculate remaining on each ball in last frame if a spare", function() {
+      lastFrame = new Frame(true);
+      expect(lastFrame.pinsRemaining()).toEqual(10);
+      lastFrame.play(5);
+      expect(lastFrame.pinsRemaining()).toEqual(5);
+      lastFrame.play(5);
+      expect(lastFrame.pinsRemaining()).toEqual(10);
+      lastFrame.play(5);
+      expect(lastFrame.pinsRemaining()).toEqual(0);
+    });
+
+    it("should be calculate remaining on each ball in last frame if 10 don't go down", function() {
+      lastFrame = new Frame(true);
+      expect(lastFrame.pinsRemaining()).toEqual(10);
+      lastFrame.play(5);
+      expect(lastFrame.pinsRemaining()).toEqual(5);
+      lastFrame.play(4);
+      expect(lastFrame.pinsRemaining()).toEqual(0);
+    });
+  });
+
+  describe("Validation - game complete", function() {
+
+    it("should not allow three balls", function() {
+      frame = new Frame();
+      frame.play("-");
+      frame.play("-");
+      expect(function() {frame.play(10);}).toThrow(new Error('This game is complete'));
+    });
+
+    it("shouldn\'t allow three balls on last frame unless strike or spare", function() {
+      lastFrame = new Frame(true);
+      lastFrame.play("-");
+      lastFrame.play("-");
+      expect(function() {lastFrame.play(10);}).toThrow(new Error('This game is complete'));
+    });
+
+    it("should allow three balls on last frame if a strike", function() {
+      lastFrame = new Frame(true);
+      lastFrame.play("X");
+      lastFrame.play("-");
+      expect(function() {lastFrame.play(0);}).not.toThrow(new Error('This game is complete'));
+    });
+
+    it("should allow three balls on last frame if a spare", function() {
+      lastFrame = new Frame(true);
+      lastFrame.play("3");
+      lastFrame.play("7");
+      expect(function() {lastFrame.play("X");}).not.toThrow(new Error('This game is complete'));
+    });
+
+    it("should allow three strikes on last frame", function() {
+      lastFrame = new Frame(true);
+      lastFrame.play("X");
+      lastFrame.play("X");
+      expect(function() {lastFrame.play("X");}).not.toThrow(new Error('This game is complete'));
+    });
+  });
+
+  describe("Validation - score values whole frame", function() {
+
+    beforeEach(function() {
+      frame = new Frame();
+      lastFrame = new Frame(true);
+    });
+
+    it("should not allow scores over 10", function() {
+      frame.play(1)
+      expect(function() {frame.play(10);}).toThrow(new Error('Score for this game should not exceed 10'));
+    });
+
+    it("should allow scores up to 20 after second ball on last frame", function() {
+      lastFrame.play(10)
```

Comment:
Are there any circumstances in which the tests on line 239 and 268 pass and this test fails?